lib/phoenix/pubsub/redis_broadcaster.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Phoenix.PubSub.RedisBroadcaster do
+  use GenServer
+  require Logger
+
+  @moduledoc """
+  Worker for pooled publishes to redis and forwarding received
+  redis messages to pg2 subscribers
+  """
+
+  @doc """
+  Starts the server
+
+  TODO document options
+  """
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts)
+  end
+
+  @doc false
+  def init(opts) do
+    Process.flag(:trap_exit, true)
+    {:ok, %{status: :disconnected, opts: opts, pid: nil}}
+  end
+
+  @doc false
+  def handle_info({:EXIT, _pid, {:connection_error, {:connection_error, :econnrefused}}}, state) do
+    {:noreply, state}
+  end
+
+  @doc """
+  Publishes message to redis with lazily established connection
+  """
+  def handle_call({:publish_to_redis, topic, msg}, _from, %{status: :disconnected, opts: opts} = state) do
+    case :eredis.start_link(opts) do
+      {:ok, pid}  ->
+        handle_publish(topic, msg, %{state | status: :connected, pid: pid})
+      {:error, reason} ->
+        Logger.error fn -> "#{inspect __MODULE__} unable to connect to redis #{inspect reason}" end
+        {:reply, {:error, reason}, state}
+    end
+  end
+  def handle_call({:publish_to_redis, topic, msg}, _from, state) do
+    handle_publish(topic, msg, state)
+  end
+
+  @doc """
+  Stops the server
+  """
+  def handle_call(:stop, _from, pid) do
+    case :eredis_client.stop(pid) do
```

Comment:
This will make you connect just to stop the connection. It feels this clause should be moved up and also account for already disconnected cases (where stop can likely raise). Also, it is likely this code won't work at all because pid is a map. :D